packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -0,0 +1,745 @@
+   * @throws {WalletError} if another error occurs
+   */
+  public async delete(): Promise<void> {
+    if (!this.walletConfig) {
+      throw new WalletError(
+        'Can not delete wallet that does not have wallet config set. Make sure to call create wallet before deleting the wallet'
+      )
+    }
+
+    this.logger.info(`Deleting wallet '${this.walletConfig.id}'`)
+
+    if (this._store) {
+      await this.close()
+    }
+
+    try {
+      const { uri } = uriFromWalletConfig(this.walletConfig, this.fileSystem.basePath)
+      await Store.remove(uri)
+    } catch (error) {
+      const errorMessage = `Error deleting wallet '${this.walletConfig.id}': ${error.message}`
+      this.logger.error(errorMessage, {
+        error,
+        errorMessage: error.message,
+      })
+
+      throw new WalletError(errorMessage, { cause: error })
+    }
+  }
+
+  public async export(exportConfig: WalletExportImportConfig) {
+    // TODO
+    throw new WalletError('AskarWallet Export not yet implemented')
+  }
+
+  public async import(walletConfig: WalletConfig, importConfig: WalletExportImportConfig) {
+    // TODO
+    throw new WalletError('AskarWallet Import not yet implemented')
+  }
+
+  /**
+   * @throws {WalletError} if the wallet is already closed or another error occurs
+   */
+  public async close(): Promise<void> {
+    this.logger.debug(`Closing wallet ${this.walletConfig?.id}`)
+    if (!this._store) {
+      throw new WalletError('Wallet is in invalid state, you are trying to close wallet that has no handle.')
+    }
+
+    try {
+      await this.session.close()
+      await this.store.close()
+      this._session = undefined
+      this._store = undefined
+      this.publicDidInfo = undefined
+    } catch (error) {
+      const errorMessage = `Error closing wallet': ${error.message}`
+      this.logger.error(errorMessage, {
+        error,
+        errorMessage: error.message,
+      })
+
+      throw new WalletError(errorMessage, { cause: error })
+    }
+  }
+
+  public async initPublicDid(didConfig: DidConfig) {
+    // Not implemented, as it does not work with legacy Ledger module
+  }
+
+  /**
+   * Create a key with an optional seed and keyType.
+   * The keypair is also automatically stored in the wallet afterwards
+   *
+   * @param seed string The seed for creating a key
+   * @param keyType KeyType the type of key that should be created
+   *
+   * @returns a Key instance with a publicKeyBase58
+   *
+   * @throws {WalletError} When an unsupported keytype is requested
+   * @throws {WalletError} When the key could not be created
+   */
+  public async createKey({ seed, keyType }: WalletCreateKeyOptions): Promise<Key> {
+    try {
+      if (keyTypeSupportedByAskar(keyType)) {
+        const algorithm = keyAlgFromString(keyType)
+
+        // Create key from seed
+        const key = seed
+          ? AskarKey.fromSeed({ seed: new TextEncoder().encode(seed), algorithm })
```

Comment:
text encoder is not available in RN

Reply:
I've seen that TextEncoder is being used in [a few parts ](https://github.com/hyperledger/aries-askar/blob/697a77fa71312ee2bedc414e31cb3c9ba1124ecd/wrappers/javascript/aries-askar-shared/src/store/Session.ts#L98)of aries-askar-shared. As this code is used in both NodeJS and RN I think this should be updated, right?

Reply:
Hmm yes...

Reply:
We've had this issue before, and it was doable to do it in other ways, however it'll probably be hard to do without a buffer polyfill in RN (or a text encoder polyfill)

Maybe the encoding/decoding needs to happen on the platform specific implementation?

Or we add a polyfill to both (afj already uses buffer polyfill). Thoughts?